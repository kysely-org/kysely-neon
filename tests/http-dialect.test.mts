import { beforeAll, beforeEach, describe, expect, it } from 'vitest'
import {
	initTest,
	resetState,
	SUPPORTED_DIALECTS,
	type TestContext,
} from './test-setup.mjs'

for (const dialect of SUPPORTED_DIALECTS) {
	describe(`neon: ${dialect}`, () => {
		let ctx: TestContext

		beforeAll(async () => {
			ctx = await initTest(dialect)
		})

		beforeEach(async () => {
			await resetState()
		})

		it('should execute select queries', async () => {
			const result = await ctx.db.selectFrom('person').selectAll().execute()

			expect(result).toMatchSnapshot([
				{ id: '48856ed4-9f1f-4111-ba7f-6092a1be96eb', name: 'moshe' },
				{ id: '28175ebc-02ec-4c87-9a84-b3d25193fefa', name: 'haim' },
				{ id: 'cbbffbea-47d5-40ec-a98d-518b48e2bb5d', name: 'rivka' },
			])
		})
	})
}
