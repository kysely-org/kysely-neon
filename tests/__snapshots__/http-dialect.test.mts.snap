// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`neon: http > should execute select queries 1`] = `
[
  {
    "id": "48856ed4-9f1f-4111-ba7f-6092a1be96eb",
    "name": "moshe",
  },
  {
    "id": "28175ebc-02ec-4c87-9a84-b3d25193fefa",
    "name": "haim",
  },
  {
    "id": "cbbffbea-47d5-40ec-a98d-518b48e2bb5d",
    "name": "rivka",
  },
]
`;

exports[`neon: ws > should execute select queries 1`] = `
[
  {
    "id": "48856ed4-9f1f-4111-ba7f-6092a1be96eb",
    "name": "moshe",
  },
  {
    "id": "28175ebc-02ec-4c87-9a84-b3d25193fefa",
    "name": "haim",
  },
  {
    "id": "cbbffbea-47d5-40ec-a98d-518b48e2bb5d",
    "name": "rivka",
  },
]
`;
